Slim/Plugin/Podcast/Plugin.pm

```diff
@@ -328,14 +321,26 @@ sub searchHandler {
 }
 
 sub registerProvider {
-	my ($class, $name, $provider, $force) = @_;
+	my ($class, $force) = @_;
 
-	if (!$providers{$name} || $force) {
-		$providers{$name} = bless $provider, $class;
+	eval "require $class";
+
+	# in case somebody provides a faulty plugin
+	if ($@) {
+		$log->warn("cannot load $class");
+		return;
 	}
-	else {
-		$log->warn(sprintf('Podcast aggregator %s is already registered!', $name));
+
+	my $name = $class->getName;
+
+	# load if not already there or forced
+	if (!$providers{$name} || $force) {
```

Comment:
When to use The Force, ask I do.

Reply:
:smile:

Reply:
or do you really want to name that arg doForce?

Reply:
I'm only wondering what it does, and when it's being used.

Reply:
Ok, just saw your extension. You need it to override the existing implementation. Got it.